src/Build/Graph/ProjectGraph.cs

```diff
@@ -0,0 +1,34 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Build.Exceptions;
+
+namespace Microsoft.Build.Graph
+{
+    /// <summary>
+    /// Represents a graph of evaluated projects.
+    /// </summary>
+    internal class ProjectGraph
```

Comment:
Any preference to class vs interface? Interfaces may make for better mocking for UTs, but comes at the cost of another layer of abstraction. Realistically there would probably not be multiple impls of this interface.